mindmeld/augmentation.py

```diff
@@ -0,0 +1,285 @@
+            logger.warning(
+                """'paths' field is not configured or misconfigured in the `config.py`.
+                 Can't find files to augment."""
+            )
+            return
+
+        rules = config["paths"]
+
+        filtered_paths = []
+
+        for rule in rules:
+            pattern = _get_pattern(rule)
+            compiled_pattern = re.compile(pattern)
+            filtered_paths.extend(
+                self._resource_loader.filter_file_paths(
+                    compiled_pattern=compiled_pattern, file_paths=all_file_paths
+                )
+            )
+        return filtered_paths
+
+    def _read_path_queries(self, path):
+        """Returns all queries in a specified file path.
+
+        Args:
+            path (str): File path.
+
+        Return:
+            queries (list): List of queries in the file path.
+        """
+        with open(path, "r") as f:
+            queries = f.readlines()
+        return queries
+
+    def _write_files(self, path, augmented_queries):
+        """Writes augmented queries to a new file in the path.
+
+        Args:
+            path (str): File path to the original file.
+            augmented_queries (list): List of augmented queries returned by class.
+        """
+        write_path = path.strip(".txt") + ".augmented.txt"
+
+        with open(write_path, "w") as outfile:
+            for query in augmented_queries:
+                outfile.write(query.strip("\n") + "\n")
+
+
+class EnglishParaphraser(Augmentor):
+    """Paraphraser class for generating English paraphrases."""
+
+    def __init__(self, app_path, config=None):
+        """Initializes an English paraphraser.
+
+        Args:
+            app_path (str): The location of the MindMeld app
+            config (dict, optional): A config object to use. This will
+                override the config specified by the app's config.py file.
+        """
+        super().__init__(app_path=app_path, config=config)
+
+        model_name = "tuner007/pegasus_paraphrase"
+        torch_device = "cuda" if torch.cuda.is_available() else "cpu"
+        self.tokenizer = PegasusTokenizer.from_pretrained(model_name)
+        self.model = PegasusForConditionalGeneration.from_pretrained(
+            model_name, force_download=False
+        ).to(torch_device)
+
+    def _get_response(self, query, num_return_sequences=10, num_beams=10):
+        """Generates paraphrase responses for given query.
+
+        Args:
+            query (str): An application query.
+            num_return_sequences (int, optional): Maximum number of paraphrases to be generated.
+            num_beams (int, optional):
+
+        Return:
+            paraphrases (list(str)): List of paraphrased queries.
+        """
+        batch = self.tokenizer.prepare_seq2seq_batch(
+            [query], truncation=True, padding="longest", max_length=60
+        )
+        translated = self.model.generate(
+            **batch,
+            max_length=60,
```

Comment:
add these params as constants if they are not configured

Reply:
Added as default params in the constructors of respective paraphrasers, as discussed.